System.NullReferenceException with mono 5.0

**CKAN Version:** 
1.22.1-0-g174271a

**Operating System:** 
Debian GNU/Linux 8

**The issue you are experiencing:** 
CKAN crashes every time I press Download or Apply (with new mod selected) with error below.

**How to recreate this issue:**
Ussue seems to be caused by latest upgrade of mono to 5.0.
After I downgraded mono to 4.8 everything is fine again.

**CKAN error codes (if applicable):** 
Unhandled exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at (wrapper managed-to-native) CurlSharp.NativeMethods:curl_easy_perform (intptr)
  at CurlSharp.CurlEasy.Perform () [0x00007] in <db7f644cef3d44268a3a52ddab219b4b>:0 
  at CKAN.NetAsyncDownloader.CurlWatchThread (System.Int32 index, CurlSharp.CurlEasy easy, System.IO.FileStream stream) [0x00010] in <db7f644cef3d44268a3a52ddab219b4b>:0 
  at CKAN.NetAsyncDownloader+<DownloadCurl>c__AnonStorey1.<>m__1 () [0x00001] in <db7f644cef3d44268a3a52ddab219b4b>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <4dc8ec68b0964e099af86e50301f5f3c>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <4dc8ec68b0964e099af86e50301f5f3c>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <4dc8ec68b0964e099af86e50301f5f3c>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <4dc8ec68b0964e099af86e50301f5f3c>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <4dc8ec68b0964e099af86e50301f5f3c>:0

@politas I still get the same error.

```
> ckan.exe version
v1.22.2-dev+6b36748aa247
> ckan.exe
7344 [Threadpool worker] WARN CKAN.Curl (null) - Curl environment not pre-initialised, performing non-threadsafe init.
Stacktrace:

  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) CurlSharp.NativeMethods.curl_easy_perform (intptr) [0x00002] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0
  at CurlSharp.CurlEasy.Perform () [0x00007] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0
  at CKAN.Net.DownloadText (string) [0x00060] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0
  at CKAN.Net.DownloadText (System.Uri) [0x00006] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0
  at CKAN.GameVersionProviders.KspBuildMap.TrySetRemoteBuildMap () [0x00000] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0
  at CKAN.GameVersionProviders.KspBuildMap.Refresh (bool) [0x00015] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0
  at CKAN.GameVersionProviders.KspBuildMap.Refresh () [0x00000] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0
  at CKAN.Repo.UpdateRegistry (System.Uri,CKAN.Registry,CKAN.KSP,CKAN.IUser,bool) [0x0000a] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0
  at CKAN.Repo.UpdateAllRepositories (CKAN.RegistryManager,CKAN.KSP,CKAN.IUser) [0x00048] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0
  at CKAN.Main.UpdateRepo (object,System.ComponentModel.DoWorkEventArgs) [0x00012] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0
  at System.ComponentModel.BackgroundWorker.OnDoWork (System.ComponentModel.DoWorkEventArgs) [0x00019] in <2b0d86369d72459baed0cee98a8e578a>:0
  at System.ComponentModel.BackgroundWorker.WorkerThreadStart (object) [0x0000e] in <2b0d86369d72459baed0cee98a8e578a>:0
  at (wrapper runtime-invoke) <Module>.runtime_invoke_void__this___object (object,intptr,intptr,intptr) [0x0001e] in <2b0d86369d72459baed0cee98a8e578a>:0
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Runtime.Remoting.Messaging.AsyncResult.Invoke (System.Runtime.Remoting.Messaging.AsyncResult) [0x0000c] in <a07d6bf484a54da2861691df910339b1>:0
  at System.Runtime.Remoting.Messaging.AsyncResult.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00000] in <a07d6bf484a54da2861691df910339b1>:0
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <a07d6bf484a54da2861691df910339b1>:0
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <a07d6bf484a54da2861691df910339b1>:0
  at (wrapper runtime-invoke) <Module>.runtime_invoke_bool (object,intptr,intptr,intptr) [0x0001e] in <a07d6bf484a54da2861691df910339b1>:0
/proc/self/maps:
00400000-00799000 r-xp 00000000 08:15 399231                             /usr/bin/mono-sgen
... <shortened> ...
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]

Native stacktrace:

7572 [Threadpool worker] WARN CKAN.GameVersionProviders.KspBuildMap (null) - Could not retrieve latest build map from: https://raw.githubusercontent.com/KSP-CKAN/CKAN-meta/master/builds.json
15330 [1] ERROR CKAN.ErrorDialog (null) - Unhandled exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at (wrapper managed-to-native) CurlSharp.NativeMethods:curl_easy_perform (intptr)
  at CurlSharp.CurlEasy.Perform () [0x00007] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0 
  at CKAN.NetAsyncDownloader.CurlWatchThread (System.Int32 index, CurlSharp.CurlEasy easy, System.IO.FileStream stream) [0x0000f] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0 
  at CKAN.NetAsyncDownloader+<>c__DisplayClass16_0.<DownloadCurl>b__1 () [0x00000] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <a07d6bf484a54da2861691df910339b1>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <a07d6bf484a54da2861691df910339b1>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <a07d6bf484a54da2861691df910339b1>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <a07d6bf484a54da2861691df910339b1>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <a07d6bf484a54da2861691df910339b1>:0  
<now the stacktrace appears in a small dialogue window in GUI, the rest follows after I dismiss the window>
[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
  at (wrapper managed-to-native) CurlSharp.NativeMethods:curl_easy_perform (intptr)
  at CurlSharp.CurlEasy.Perform () [0x00007] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0 
  at CKAN.NetAsyncDownloader.CurlWatchThread (System.Int32 index, CurlSharp.CurlEasy easy, System.IO.FileStream stream) [0x0000f] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0 
  at CKAN.NetAsyncDownloader+<>c__DisplayClass16_0.<DownloadCurl>b__1 () [0x00000] in <ef6d5dff235b44eeb42bfd227d9f90f5>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <a07d6bf484a54da2861691df910339b1>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <a07d6bf484a54da2861691df910339b1>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <a07d6bf484a54da2861691df910339b1>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <a07d6bf484a54da2861691df910339b1>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <a07d6bf484a54da2861691df910339b1>:0
```

--- a/Core/Net/NetAsyncDownloader.cs
+++ b/Core/Net/NetAsyncDownloader.cs
@@ -84,7 +84,7 @@ namespace CKAN
             }
 
             // adding chicken bits
-            if (Platform.IsWindows || System.Environment.GetEnvironmentVariable("KSP_CKAN_USE_CURL") != null) {
+            if (Platform.IsWindows || System.Environment.GetEnvironmentVariable("KSP_CKAN_USE_CURL") == null) {
                 DownloadNative();
             }
             else

